DOC-8818: Added note on Eventing Service not covered by node-to-node all

tonyjhillman opened the pull request.

tonyjhillman committed:
DOC-8818: Added note on Eventing Service not covered by node-to-node all

tonyjhillman committed:
DOC-8818: Specified that n2n for Eventing is available in 7.0.2

jon-strabala commented:
LGTM

RayOffiah commented on modules/learn/pages/clusters-and-availability/node-to-node-encryption.adoc:
```diff
@@ -17,6 +17,9 @@ When node-to-node encryption is first enabled, this is the default.
 Node-to-node encryption can be supported by means of either the Couchbase-provided certificates with which nodes are protected by default; or certificates installed by the administrator.
++
+Note, however, that node-to-node encryption has _not_ been implemented for the Eventing Service in versions of Couchbase Server prior to 7.0.2.
+Therefore, on such earlier versions, when _all_ is specified, whereas the data passed to and from all other Couchbase Services is passed in encrypted form, the data passed to and from the Eventing Service continues to be passed in _unencrypted_ form.
```
```suggestion
Therefore, on earlier versions, when _all_ is specified, the data passed to and from all other Couchbase Services is passed in encrypted form, whereas the data passed to and from the Eventing Service continues to be passed in _unencrypted_ form.
```

jon-strabala replied:
Sorry for the late update here ...  as such there will be a bit more rewrite.
1. Eventing supports node-to-node encryption for starting in version 6.6.5 and also starting in version 7.02
2. Note that all Eventing Functions must be paused or undeployed when enabling TLS to prevent possible mutation loss.

Item 1 is due to the back porting node-to-node encryption for Eventing to 6.6.5 _Refer to MB-48048. [B.P. to 6.6.4] Backport of Enforce TLS - Eventing._  So 7.0.0 and 7.0.1 don't support encryption as such we need to discuss both version trains.

Item 2 occurs because on an upgrade to TLS KV will close and restart the DCP streams to enable TLS thus any buffered events/mutations would be dropped.

jon-strabala committed:
Clarify Eventing TLS versions and best practices

My attempt to finalize this page (should include Ray's change too) to clarify the two (2) software trains or branches when Eventing  supported encrypted communication.

Additionally indicated that to avoid mutation loss to the Eventing service that all deployed functions need to be temporarily paused on an encryption change.

tonyjhillman commented:
I'm closing this and restarting under DOC-9619.
